pkg/utils/tests/logs.go

```diff
@@ -0,0 +1,56 @@
+package tests
+
+import (
+	"strings"
+	"testing"
+
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// WaitForLogMessage waits until at least one log message containing the
+// specified msg is emitted.
+// NOTE: This does not "pop" messages so it cannot be used multiple times to
+// check for new instances of the same msg. See WaitForLogMessageCount instead.
+//
+// Get a *observer.ObservedLogs like so:
+//
+//	observedZapCore, observedLogs := observer.New(zap.DebugLevel)
+//	lggr := logger.TestLogger(t, observedZapCore)
+func WaitForLogMessage(t *testing.T, observedLogs *observer.ObservedLogs, msg string) {
```

Comment:
Maybe these should be renamed for consistency too?
```suggestion
func AssertLogMessageEventually(t *testing.T, observedLogs *observer.ObservedLogs, msg string) {
```
```go
AssertLogMessageCountEventually
```
Kinda verbose though :shrug:

Reply:
AssertLogEventually or AssertLogMsgEventually?

Reply:
Also we'll have to do the same for WaitForLogMessageCount